labellines/line_label.py

```diff
@@ -0,0 +1,202 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+import matplotlib.patheffects as patheffects
+import numpy as np
+from matplotlib.text import Text
+
+if TYPE_CHECKING:
+    from datetime import datetime
+    from typing import Any, Union
+
+    from matplotlib.axes import Axes
+    from matplotlib.lines import Line2D
+
+    Position = Union[float, datetime, np.datetime64]
+    ColorLike = Any  # mpl has no type annotations so this is just a crutch
+    # Once support for python <3.8 is dropped this should be Literal["auto"]
+    AutoLiteral = str
+
+
+class LineLabel(Text):
+    """This artist adds a label onto a preexisting Line2D object"""
+
+    _line: Line2D
+    """Annotated line"""
+
+    _target_x: Position
+    """Requested x position of the label, as supplied by the user"""
+
+    _ax: Axes
+    """Axes containing the line"""
+
+    _auto_align: bool
+    """Align text with the line (True) or parallel to x axis (False)"""
+
+    _yoffset: float
+    """An additional y offset for the label"""
+
+    _yoffset_logspace: bool
+    """Sets whether to treat _yoffset exponentially"""
+
+    _label_pos: np.ndarray
+    """Position of the label, computed from _target_x and line data"""
+
+    _anchor_a: np.ndarray
+    """Anchor A for rotation calculation, point of _line neighbouring this label"""
+
+    _anchor_b: np.ndarray
+    """Anchor B for rotation calculation, point of _line neighbouring this label"""
+
+    def __init__(
+        self,
+        line: Line2D,
+        x: Position,
+        label: str | None = None,
+        align: bool = True,
+        yoffset: float = 0,
+        yoffset_logspace: bool = False,
+        outline_color: AutoLiteral | ColorLike | None = "auto",
+        outline_width: float = 8,
+        **kwargs,
+    ) -> None:
+        """
+
+        Parameters
+        ----------
+        line : Line2D
+            Line to be decorated.
+        x : Position
+            Horizontal target position for the label (in data units).
+        label : Optional[str], optional
```

Comment:
I now `Optional[str]` is indeed the type, but I'd rather write
```suggestion
        label : str, optional
```
or
```suggestion
        label : str | None, optional
```
Also the docstrings in the Artist do not match anymore those in the `labelLines` function. Would it be possible to homogeneise them?